### PR TITLE
feat(settings): add save feedback for preferences changes

### PR DESCRIPTION
**Pull request by lspassos1**
## Summary

Reapplies the settings save feedback work from #1269 on top of current `main`, without carrying the stale branch drift that is now breaking that PR.

## Root cause

The app already had inline `Saved` feedback for the Panels tab, but Preferences changes (theme, map provider, language, AI toggles, etc.) still had no equivalent confirmation. The older PR branch also diverged badly from `main` and no longer typechecked cleanly.

## Changes

- add a shared global toast helper in `src/utils/toast.ts` with `role="status"`
- wire `renderPreferences()` to accept `onSettingSaved` and fire it once after persisted preference changes
- show the global `Saved` toast only for Preferences changes
- keep the Panels tab on its existing inline `Saved` status, so there is no duplicate feedback there
- replace duplicate body-level toast implementations in `UnifiedSettings`, `event-handlers`, and `country-intel`
- add a guardrail test covering the shared toast path and the no-duplicate Panels behavior

## Validation

- `npx tsx --test tests/settings-save-feedback.test.mjs`
- `npm run typecheck`
- `npx biome lint src/utils/toast.ts src/components/UnifiedSettings.ts src/services/preferences-content.ts src/app/event-handlers.ts src/app/country-intel.ts tests/settings-save-feedback.test.mjs`

## Risk

Low risk. The change is limited to settings-related feedback paths and a shared toast helper, with the Panels tab intentionally left on its existing inline status behavior.

Ref #1247
Supersedes #1269

### CODE REVIEW

**Comment by SebastienMelki** [comment]
@lspassos1 — Greptile rated this 5/5 (safe to merge) and the architecture is clean: shared toast.ts utility, onSettingSaved callback, removes 3 duplicate implementations. Branch just needs a rebase on main to resolve conflicts.

Two non-blocking notes:
1. Toast timeout silently changed from 4000ms to 3000ms — intentional?
2. Test regex depends on exact 2-space indentation — a simpler string check would be more robust.

@koala73 — recommending merge after rebase.

**Comment by SebastienMelki** [comment]
Note: merge conflicts with main. Fork PR — can't rebase from maintainer side. @lspassos1 — please rebase on current main. Greptile rated this 5/5, so once rebased it should be quick to merge.

**Comment by lspassos1** [comment]
Rebased on current `main` and addressed the two follow-up notes from @SebastienMelki  /Greptile in the latest push.

What changed:
- kept the shared helper in `src/utils/toast.ts` as `showToast(message, durationMs = 3000)`
- preserved the previous 4000ms behavior for the two `UnifiedSettings`-originated toasts by passing `4000` explicitly from `src/components/UnifiedSettings.ts`
- replaced the regex-based `savePanelChanges()` assertion in `tests/settings-save-feedback.test.mjs` with a method-body extractor that walks braces instead of depending on closing-brace indentation

Validation:
- `npx tsx --test tests/settings-save-feedback.test.mjs`

The Panels tab still stays on inline `Saved` status only; no duplicate global toast was added there.

File: tests/settings-save-feedback.test.mjs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  });
+
+  it('keeps Panels save on inline status only', () => {
+    const saveMatch = settingsSrc.match(/private savePanelChanges\(\): void \{([\s\S]*?)\n {2}\}/);
+    assert.ok(saveMatch, 'savePanelChanges() not found');
+    assert.doesNotMatch(saveMatch[1], /showToast\(/);
+  });
+
+  it('removes duplicate global toast implementations from event handlers and country intel', () => {
+    assert.match(handlersSrc, /import \{ showToast \} from '@\/utils\/toast';/);
```
[the remark this answers is not in the document]
Comment (lspassos1) [reply tests/settings-save-feedback.test.mjs:58]: Addressed in the latest push. The guardrail no longer depends on the closing-brace indentation; it now locates `savePanelChanges()` by signature and extracts the body by walking brace depth before asserting that `showToast(` is absent.

File: src/utils/toast.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  toast.className = 'toast-notification';
+  toast.setAttribute('role', 'status');
+  toast.textContent = message;
+
+  document.body.appendChild(toast);
+  requestAnimationFrame(() => toast.classList.add('visible'));
+  setTimeout(() => {
+    toast.classList.remove('visible');
+    setTimeout(() => toast.remove(), 300);
+  }, 3000);
```
[the remark this answers is not in the document]
Comment (lspassos1) [reply src/utils/toast.ts]: Addressed in the latest push. The shared helper keeps a 3000ms default, but the two `UnifiedSettings` call sites that previously used 4000ms now pass `4000` explicitly, so the modal-originated timeout behavior is preserved.

**Comment by lspassos1** [comment]
Superseded by #7403, which ports this work directly onto current upstream `main`.

The replacement preserves the intent and all prior review decisions:

- shared accessible body-level toast utility
- one save confirmation for persisted Preferences changes
- Panels keeps its existing inline status only
- import/export keeps its separate scoped feedback
- the previous 4-second/3-second durations are preserved
- the indentation-sensitive test was replaced with brace-walking extraction
- later settings work on `main`, including font scale, remains intact
- the newer `CountryIntelManager.showToast` test seam is preserved as a one-line delegate to the shared utility, without duplicating DOM logic

All required CI gates on #7403 are green, including unit, DOM, typecheck, Biome, sidecar, full-variant smoke, security audit, and Vercel preview.

Closing this stale fork PR avoids keeping two implementations of the same feature open.
